Remove regex lookbehind

Samuel-Lewis opened the pull request:
Resolves #14 by having broader grouping and just getting the specific match.

Also fixed the tests, because some of the target videos no longer exist. `ts-mocha` was also missing from the dev deps, so I threw that in there as well.

Tenpi reviewed (changes requested):
First I want to note that this is not safe to use on the front-end because anyone could see your API key, but I'm not against it.

The regex you are using looks prone to breaking so I'd prefer it if you take the first capture group and remove the front part, like
```ts
id = String(resolvable.match(/(channel\/)(.*?)(?=(?:\?|\/|$))/)[0]).replace("channel/", "")
```

Also, thanks for using one of my videos for the test I didn't expect that.

Samuel-Lewis commented:
lol, in case you can't tell, I'm not great at git 😛

Fixed my conflicts and changed to the pattern matching you described. All tests pass with it.

> this is not safe to use on the front-end because anyone could see your API key

Yeah, I've been wrestling with this. You can set the API key to only accept requests from certain domains, which is good enough for my toy project. I'm aware it's not a perfect solution and still able to be spoofed, but I'll work that one out if this goes beyond a toy project.

> thanks for using one of my videos for the test I didn't expect that

Good work should be recognised 😄

Tenpi commented:
> Yeah, I've been wrestling with this. You can set the API key to only accept requests from certain domains, which is good enough for my toy project. I'm aware it's not a perfect solution and still able to be spoofed, but I'll work that one out if this goes beyond a toy project.

Typically the way you should use this is by sending an API request from the front-end to your server; you use this package on the server side and then send the response back to the front-end. In this manner your API key is never exposed.

Alternatively, you could use a library which uses web scraping because that wouldn't need an API key. If this is just a small personal project I suppose you don't have to worry too much, just know it's insecure

This should be available on `v0.2.5`
